extensions/azcli/src/azdata.ts

```
@@ -262,451 +240,76 @@ export class AzdataTool implements azdataExt.IAzdataApi {
- * @param raw The raw version output from azdata --version
+ * Parses out the Azure CLI version from the raw az version output
+ * @param raw The raw version output from az --version
  */
 function parseVersion(raw: string): string {
 	// Currently the version is a multi-line string that contains other version information such
-	// as the Python installation, with the first line being the version of azdata itself.
-	const lines = raw.split(os.EOL);
-	return lines[0].trim();
+	// as the Python installation, with the first line holding the version of az itself.
+	const start = raw.search('azure-cli');
+	const end = raw.search('core');
+	raw = raw.slice(start, end).replace('azure-cli', '');
+	return raw.trim();
 }
 
 /**
- * Gets the latest azdata version for MacOs clients
+ * Parses out the arcdata extension version from the raw az version output
+ * @param raw The raw version output from az --version
  */
-async function discoverLatestStableAzdataVersionDarwin(): Promise<SemVer> {
-	// set brew tap to azdata-cli repository
-	await executeCommand('brew', ['tap', 'microsoft/azdata-cli-release']);
-	await executeCommand('brew', ['update']);
-	let brewInfoAzdataCliJson;
-	// Get the package version 'info' about 'azdata-cli' from 'brew' as a json object
-	const brewInfoOutput = (await executeCommand('brew', ['info', 'azdata-cli', '--json'])).stdout;
-	try {
-		brewInfoAzdataCliJson = JSON.parse(brewInfoOutput);
-	} catch (e) {
-		throw Error(`failed to parse the JSON contents output of: 'brew info azdata-cli --json', text being parsed: '${brewInfoOutput}', error:${getErrorMessage(e)}`);
+function parseArcExtensionVersion(raw: string): string {
+	// Currently the version is a multi-line string that contains other version information such
+	// as the Python installation and any extensions.
+	const start = raw.search('arcdata');
+	if (start === -1) {
+		vscode.window.showErrorMessage(loc.arcdataExtensionNotInstalled);
```

Just showing an error? Usually you'd want to throw an exception or something I would think so that the code can handle the installation being "invalid"